### PR TITLE
Shorten URL for link definitions

### LINKED ISSUE #4853: Markdown: links without title could display shortened host name
### Is your feature request related to a problem?

Today, for link without title, https://www.example.com/index.html, we display "www.example.com". In JS, `new URL(url).host`.

We can add special treatments to remove leading "www." if they are on standard port number. See additional context.

### Describe the suggestion or request in detail

_No response_

### Describe alternatives you have considered

_No response_

### Additional context

| URL | Display name | Treated |
| - | - | - |
| `https://www.example.com/index.html` | [example.com](https://www.example.com/index.html) | Yes |
| `https://www.example.co.jp/index.html` | [example.co.jp](https://www.example.co.jp/index.html) | Yes |
| `https://www.example.com:443/index.html` | [example.com](https://www.example.com:443/index.html) | Yes, default port number |
| `http://www.example.com:80/index.html` | [example.com](http://www.example.com:80/index.html) | Yes, default port number |
| `https://www.support.example.com/index.html` | [support.example.com](https://www.support.example.com/index.html) | Yes, edge case, difficult to deal with gTLD |
| `https://www.microsoft/index.html` | [microsoft](https://www.microsoft/index.html) | Yes, edge case, `microsoft` is gTLD, difficult to deal with gTLD |
| `https://www.例子.com/index.html` | [xn--fsqu00a.com](https://www.例子.com/index.html) | Yes, we could consider un-punycode it |
| `https://www.example.com:8443/index.html` | [www.example.com:8443](https://www.example.com:8443/index.html) | No, not default port number |
| `https://support.example.com/index.html` | [support.example.com](https://support.example.com/index.html) | No, not leading with "www.' |
| `https://www2.example.com/index.html` | [www2.example.com](https://www2.example.com/index.html) | No, not leading with "www.' |

https://msazure.visualstudio.com/CCI/_workitems/edit/25112776

### PR DESCRIPTION
**Pull request by compulim**
> Fixes #4853.

### Added

-  Resolves [#4853](https://github.com/microsoft/BotFramework-WebChat/issues/4853). Shorten URLs in link definitions UI, by [@compulim](https://github.com/compulim), in PR [#4860](https://github.com/microsoft/BotFramework-WebChat/pull/4860)

## Description

URLs without title will be shorten for readability. See #4853 for rules.

## Design

If the URL is intuitive enough, we will shorten it.

Signals of intuitiveness: http/https, default protocol, starts with `www`.

## Specific Changes

- Added `shortenURL` to shorten URLs
- Updated link definition item to shorten URL if title is not available

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [x] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [x] ~Accessibility reviewed (tab order, content readability, alt text, color contrast)~
-  [x] ~Browser and platform compatibilities reviewed~
-  [x] ~CSS styles reviewed (minimal rules, no `z-index`)~
-  [x] ~Documents reviewed (docs, samples, live demo)~
-  [x] ~Internationalization reviewed (strings, unit formatting)~
-  [x] ~`package.json` and `package-lock.json` reviewed~
-  [x] ~Security reviewed (no data URIs, check for nonce leak)~
-  [x] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/component/src/Attachment/Text/private/shortenURL.spec.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  ['https://www.example.com/', 'example.com'],
+  ['https://www.example.com/index.html', 'example.com'],
+  ['HTTPS://WWW.EXAMPLE.COM/index.html', 'example.com'],
+  ['https://www.example.co.jp/index.html', 'example.co.jp'],
+  ['https://www.example.com:443/index.html', 'example.com'],
+  ['http://www.example.com:80/index.html', 'example.com'],
+  ['https://www.support.example.com/index.html', 'support.example.com'],
+  ['https://www.microsoft/index.html', 'microsoft'],
+  ['https://www.例子.com/index.html', 'xn--fsqu00a.com'],
+  ['https://www.example.com:8443/index.html', 'www.example.com:8443'],
```
Comment (jonchurch) [inline packages/component/src/Attachment/Text/private/shortenURL.spec.ts:13]: If the url comes in as https://www.example.com:8080 would we want to strip the protocol?

Comment (jonchurch) [reply packages/component/src/Attachment/Text/private/shortenURL.spec.ts:13]: rules say:


https://www.example.com:8443/index.html | www.example.com:8443 | No, not default port number

Suggesting default ports would be stripped?

File: packages/component/src/Attachment/Text/private/shortenURL.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export default function shortenURL(urlString: string): string {
```
Comment (jonchurch) [inline packages/component/src/Attachment/Text/private/shortenURL.ts:1]: Would `extractHostnameWithSubdomain` be more accurate?

File: packages/component/src/Attachment/Text/private/shortenURL.spec.ts (hunk 1 above)
Comment (compulim) [reply packages/component/src/Attachment/Text/private/shortenURL.spec.ts:13]: No, `https://www.example.com:8080/` will become `www.example.com:8080`, not stripping the port.

File: packages/component/src/Attachment/Text/private/shortenURL.ts (hunk 2 above)
Comment (compulim) [reply packages/component/src/Attachment/Text/private/shortenURL.ts:1]: Updating it.

**Review by beyackle2 — APPROVED** [review]

File: packages/component/src/Attachment/Text/private/shortenURL.spec.ts (hunk 1 above)
Comment (compulim) [reply packages/component/src/Attachment/Text/private/shortenURL.spec.ts]: Added a test to show the port number 8080 is not stripped.

**Review by beyackle2 — APPROVED** [review]

**Review by jonchurch — APPROVED** [review]
LGTM

**Pull request merged**